codec-http2/src/main/java/io/netty/handler/codec/http2/DefaultHttp2Connection.java

```diff
@@ -1121,19 +1121,19 @@ void addToActiveStreams(DefaultStream stream) {
 
         void removeFromActiveStreams(DefaultStream stream) {
             if (streams.remove(stream)) {
-                try {
-                    // Update the number of active streams initiated by the endpoint.
-                    stream.createdBy().numActiveStreams--;
+                // Update the number of active streams initiated by the endpoint.
+                stream.createdBy().numActiveStreams--;
+            }
+            notifyClosed(stream);
```

Comment:
@Scottmitch this is somewhat unrelated, but I noticed that this seemed wrong.  The previous logic would only let us close/remove if the stream had at one point become active.  This change lifts this restriction.